src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineParser.java

```diff
@@ -349,7 +437,50 @@ private void assertArgumentsAreValid() {
             throw new GATKException.ShouldNeverReachHereException("Should never happen",e);
         }
 
+    }
+
+    // Once all command line args have been processed, go through the argument definitions and
+    // validate any that are plugin class arguments against the controlling descriptor, trimming
+    // the list of argument definitions along the way by removing any that have not been set
+    // (so validation doesn't complain about missing required arguments for plugins that weren't
+    // specified) and throwing for any that have been set but are not allowed. Note that we don't trim
+    // the list of plugins themselves (just the argument definitions), since the plugin may contain
+    // other arguments that require validation.
+    private void validatePluginArguments() {
+        final List<ArgumentDefinition> actualArgumentDefinitions = new ArrayList<>();
+        for (final ArgumentDefinition argumentDefinition : argumentDefinitions) {
+            if (null == argumentDefinition.controllingDescriptor) {
+                actualArgumentDefinitions.add(argumentDefinition);
+            }
+            else {
```

Comment:
there's funny line breaking in this block and the one following it, we usually go with` } else {` I think.